sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2649,3 +2652,84 @@ case class Sentences(
     copy(str = newFirst, language = newSecond, country = newThird)
 
 }
+
+case class ToPrettyString(expr: Expression, timeZoneId: Option[String] = None)
+  extends UnaryExpression with ImplicitCastInputTypes with TimeZoneAwareExpression{
+  import ToPrettyString._
+
+  require(children.nonEmpty, s"$prettyName() should take at least 1 argument")
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
+    copy(timeZoneId = Option(timeZoneId))
+  override def child: Expression = expr
+  override def foldable: Boolean = child.foldable
+  override def nullable: Boolean = child.nullable
```

Comment:
please check `UnaryExpression`, `foldable/nullable` is already defined.

Reply:
yea.